app/models/knight.rb

```diff
@@ -9,7 +9,7 @@ def display
   end
 
   def is_valid_move?(board, stop)
-    (is_horizontal_move?(board, stop) || is_vertical_move?(board, stop)) && (!chess_piece || chess_piece.color != color)
+    (is_horizontal_move?(board, stop) || is_vertical_move?(board, stop)) && (is_space_occupied?(board, stop) && (is_opponent?(board, stop))
```

Comment:
I think you need another statement here. If the space is not occupied, the second half will be (false && true) which will return false to the entire line. And if the second half returns false the method will return false for trying to move to an empty space.

Reply:
After looking at the failed test, I think its because you have an extra ( in front of is_opponent? on this line

Reply:
No I don't think it will. The way its currently written is it pulls the id stored in that space, finds the piece with that id and then compares the colors. So if the space was empty it would pull 'nil' and I think would return false when trying to find a piece with an id of nil

Reply:
Okay so I just added a statement to handle an empty space.